applications/plugins/snake_game/helpers/snake_file_handler.c

```diff
@@ -0,0 +1,181 @@
+#include "snake_file_handler.h"
+
+#include <furi.h>
+#include <flipper_format/flipper_format.h>
+
+static void snake_game_close_file(FlipperFormat* file) {
+    if(file == NULL) {
+        furi_record_close(RECORD_STORAGE);
+        return;
+    }
+    flipper_format_file_close(file);
+    flipper_format_free(file);
+    furi_record_close(RECORD_STORAGE);
+}
+
+static FlipperFormat* snake_game_open_file() {
+    Storage* storage = furi_record_open(RECORD_STORAGE);
+    FlipperFormat* file = flipper_format_file_alloc(storage);
+
+    if(storage_common_stat(storage, SNAKE_GAME_FILE_PATH, NULL) == FSE_OK) {
+        if(!flipper_format_file_open_existing(file, SNAKE_GAME_FILE_PATH)) {
+            snake_game_close_file(file);
+            return NULL;
+        }
+    } else {
+        if(storage_common_stat(storage, APPS_DATA, NULL) == FSE_NOT_EXIST) {
+            if(!storage_simply_mkdir(storage, APPS_DATA)) {
+                return NULL;
```

Comment:
Going to leak file if something will go wrong here.
Whole way of working with file looks quite dangerous, please check how it's done in other apps.